### ISSUE TITLE
Reflect cancels other properties to be set

### ISSUE THREAD

**Issue by andreabosman16** [Polymer/lit-element#224]
I have the following setup

```
@property({type:String}) ae_tab?:string;
@property({type:Boolean,reflect:true}) ae_selected?:boolean;
```
when I set the selected boolean to true in HTML the ae_tab is undefined and when selected isn't set it works just fine. if I remove reflect everything works fine as well. Am I missing some kind of functionality with the reflect option?

**Comment by andreabosman16** [comment]
I am sorry for this really late reaction but I still have this problem and recreation in codepen or another platform won t succeed I cannot import the node_modules from lit-element. I hope if I explain the problem again you maybe try to recreate it your self. 

I am working with the following 

typscript 3.1.3
lit-element 0.6.2
and vscode

I have the following code 

```
import { html,property,Litelement } from "@polymer/lit-element";
 class MenuItem extends Button {

    @property({type:String,reflect:true}) value?:string;
    @property({type:Boolean,reflect:true}) selected?:boolean;
    public focused: boolean;

    constructor(){
        super();
        this.focused=false;
        this.value = ""
         this.selected = false;
    }
    connectedCallback(){
        super.connectedCallback();
        console.log(this.value);
        console.log(this.selected);
    };
    render(){
        return html `Hello World`
    }
};
window.customElements.define('ae-menuitem',MenuItem);
```
and the following html
```

<html>
    <head>
        <meta charset='UTF-8'>
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <title>Home</title>
    </head>
    <body onload="APVAERP.run()">
        <ae-menulist>
            <ae-menupop slot= "menupop">
                <ae-menuitem selected value = "hallo test menuitem1" ></ae-menuitem>
            </ae-menupop>
        </ae-menulist>
    </body>
</html>
```
I have 2 problems 

one:

When using reflect inside one of the properties the first attribute called inside the html is only set on the Menuitem class the other one is undefined and not useable.

two: 

When using typescript I get errors because of not declaring the properties inside the constructor but when I do define them I get an error at runtime inside my html. I get the following error in the debugger 
`Uncaught DOMException: Failed to construct 'CustomElement': The result must not have attributes `
If I remove them from the constructor the error is gone but I think this is not supposed to happen.
Because you need to define properties inside the constructor and not let them undefined.

I almost know for sure that the reflecting property is doing something wrong because when I remove it completely everything works as suppose to but when added on 1 or more properties everything goes wrong. It is particularly strange that when used on a web component only the first attribute uses the reflect function and the second one doesn't.
Hopefully, this is enough info to give me a response and help me because this error is very annoying.

**Comment by kevinpschaaf** [comment]
Without a reproduction, this will be hard to debug.

From LitElement 0.6.0 on, property-to-attribute reflection [is only ever run from `update()`](https://github.com/Polymer/lit-element/blob/bf298cef32ba50e3897f33358c38eafe5f608dd5/src/lib/updating-element.ts#L593), which is _always_ run a microtask after the setter.  This means LitElement should never be calling `setAttribute` from the `constructor`.

Prior to 0.6.0, property-to-attribute reflection [did run synchronously to the setter](https://github.com/Polymer/lit-element/commit/f131d96bcb57092c007c1907939d74c1fcc0f706#diff-c5f280d12b2d91f8b89fe4ac3b6313d8L400), and thus would have resulted in this error.

If you've absolutely verified the version of lit-element causing this error is 0.6.0+, you'll need to either provide a working repro, or else attach a screenshot of the stack trace leading to the error (e.g. in the debugger, break on exceptions, and take a screenshot of the call stack).

**Comment by kevinpschaaf** [comment]
If you can do one of the two (repro or call stack screenshot) please re-open.

**Comment by andreabosman16** [comment]
[Reflect_Test.zip](https://github.com/Polymer/lit-element/files/2675292/Reflect_Test.zip)
I uploaded a zip just npm install and npm test. Then open the reflect.html you will see the first attribute defined and the second undefined inside the console. I fixed this inside my other issue #289. If you follow those instructions it will print both values inside the console

**Comment by sorvell** [comment]
The zip above appears to be incomplete, but it looks like we won't need it.

We have finally understood the [underlying issue here](https://github.com/Polymer/lit-element/issues/258#issuecomment-447156886). This issue since it is now clearly a dup of https://github.com/Polymer/lit-element/issues/258.
